ci_tutorial/packages_pipeline/multi_configuration.rst

```diff
@@ -0,0 +1,195 @@
+Package pipeline: multi configuration
+=====================================
+
+In the previous section we were building just 1 configuration. This section will cover the case in which we need to build more
+than 1 configuration. We will use the ``Release`` and ``Debug`` configurations here for convenience, as it is easier to 
+follow, but in real case these configurations will be more like Windows, Linux, OSX, building for different architectures, 
+cross building, etc.
+
+Let's begin cleaning our cache and initializing only the ``develop`` repo:
+
+
+.. code-block:: bash
+
+    $ conan remove "*" -c  # Make sure no packages from last run
+    $ conan remote remove "*"  # Make sure no other remotes defined
+    # Add develop repo, you might need to adjust this URL
+    $ conan remote add develop http://localhost:8081/artifactory/api/conan/develop
+
+
+We will create the packages for the 2 configurations sequentially in our computer, but note these will typically run
+in different computers, so it is typical for CI systems to launch the builds of different configurations in parallel.
+
+.. code-block:: bash
+    :caption: Release build
+
+    $ cd ai
+    $ conan create . --build="missing:ai/*" -s build_type=Release --format=json > graph.json
+    $ conan list --graph=graph.json --graph-binaries=build --format=json > built.json
+    # Add packages repo, you might need to adjust this URL
+    $ conan remote add packages http://localhost:8081/artifactory/api/conan/packages
+    $ conan upload -l=built.json -r=packages -c --format=json > uploaded_release.json
+
+We have done a few changes and extra steps:
+
+- First step is similar to the one in the previous section, a ``conan create``, just making it explicit our configuration
+  ``-s build_type=Release`` for clarity, and capturing the output of the ``conan create`` in a ``graph.json`` file.
+- The second step is create from the ``graph.json`` a ``built.json`` **package list** file, with the packages that needs to be uploaded,
+  in this case, only the packages that have been built from source (``--graph-binaries=build``) will be uploaded. This is
+  done for efficiency and faster uploads.
+- Third step is to define the ``packages`` repository
+- Finally, we will upload the ``built.json`` package list to the ``packages`` repository, creating the ``uploaded_release.json``
+  package list with the new location of the packages (the server repository).
+
+Likewise, the Debug build will do the same steps:
+
+
+.. code-block:: bash
+    :caption: Debug build
+
+    $ conan create . --build="missing:ai/*" -s build_type=Debug --format=json > graph.json
```

Comment:
Pre exporting recipes might help in some scenarios like reducing concurrency issues if building in parallel in the same machine, and it can also help when there are changes in multiple repositories/packages simultaneously. In this case, we are using the basic case of only ``ai`` having changes, so a ``conan create`` should be good enough, it would be simpler for readers.